### ISSUE TITLE
typo not being caught in Stack Parameters

### ISSUE THREAD

**Issue by dhensen** [aws-cloudformation/cfn-lint#2416]
### CloudFormation Lint Version

0.67.0

### What operating system are you using?

WSL2

### Describe the bug

Type is not getting caught inside stack parameters:

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: 'Typo not caught by cfn-lint'

Parameters:
  WakandaBucket:
    Type: String
    Description: Something.

Resources:
  WakandaStack:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: "https://somewhere/foo.yaml"
      Parameters:
        Foobar: !Red WakandaBucket

  Wakanda4EverBucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: !Ref WakandaBucket
```

>Note that after Foobar comes `!Red` insead of `!Ref`.

cfn-lint output for above template is empty, exitcode 0.

When I change the BucketName !Ref to !Red, that one gives me a proper error. But not for the Stack Parameters.

### Expected behavior

I expect something like `Property X/Y/Z has an illegal function Fn::Red`

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: 'Typo not caught by cfn-lint'

Parameters:
  WakandaBucket:
    Type: String
    Description: Something.

Resources:
  WakandaStack:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: "https://somewhere/foo.yaml"
      Parameters:
        Foobar: !Red WakandaBucket

  Wakanda4EverBucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: !Ref WakandaBucket
```

**Comment by PatMyron** [comment]
reproduced with another property-type that was also a map of strings as well
